OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs

```diff
@@ -199,14 +209,19 @@ void INotifyKilled.Killed(Actor self, AttackInfo e)
 			{
 				defenderStats.ArmyValue -= cost;
 				includedInArmyValue = false;
+				playerStats.Units[info.OverrideActor ?? self.Info.Name].Count--;
```

Comment:
Might be worth pulling `info.OverrideActor ?? self.Info.Name` into its own variable in the ctor to reduce duplication.